File: src/app/document.app.ts
import { Document } from "../model/document";
import { addDocument, printDocumentGroupByType, removeDocument } from "../service/document.service";
import { myPrompt } from "../util/my.prompt.util";
import { bar } from "../view/bar.view";
import { printMenu } from "../view/menu.view";

export const runDocumentApp = async (documentList: Document[]) => {
  let conditionRunning: boolean = true;
  while (conditionRunning) {
    bar("Document");
    printMenu("Status", "Add", "Remove", "Exit");
    let input = await myPrompt("Choice: ");
    let choice: number = Number(input);
    switch (choice) {
      case 1:
        bar("Status");
        printDocumentGroupByType(documentList);
        break;
      case 2:
        bar("Add");
        await addDocument(documentList)
          .then((res) => console.log(res))
          .catch((err) => console.log(err.message));
        break;
      case 3:
        bar("Remove");
        await removeDocument(documentList)
          .then((res) => console.log(res))
          .catch((err) => console.log(err.message));
        break;
      case 4:
        conditionRunning = false;
        break;
      default:
        console.log("RETRY");
    }
  }
};

File: src/app/DocumentApp.java
package app;

import model.Document;
import service.DocumentService;
import view.BarView;
import view.MenuView;

import java.util.List;
import java.util.Scanner;

public class DocumentApp {
    public static void run(List<Document> documentList, Scanner sc) {
        boolean conditionRunning = true;
        while (conditionRunning) {
            BarView.bar("Document");
            MenuView.printMenu("Status", "Add", "Remove", "Exit");
            while (!sc.hasNextInt()) {
                System.out.println("Please enter a valid number");
                sc.next();
            }
            int choice = sc.nextInt();
            switch (choice) {
                case 1:
                    BarView.bar("Status");
                    DocumentService.printDocumentGroupByType(documentList);
                    break;
                case 2:
                    BarView.bar("Add");
                    DocumentService.addDocument(documentList, sc);
                    break;
                case 3:
                    BarView.bar("Remove");
                    DocumentService.removeDocument(documentList, sc);
                    break;
                case 4:
                    conditionRunning = false;
                    System.out.println("Bye");
                    break;
                default:
                    System.out.println("RETRY");
            }
        }
    }
}
